hearpreprocess/pipeline.py

```diff
@@ -530,6 +522,69 @@ def split_train_test_val(self, metadata: pd.DataFrame):
         metadata.loc[metadata["split_key"].isin(test_split_keys), "split"] = "test"
         return metadata
 
+    def assert_correct_kfolds(self, metadata: pd.DataFrame):
+        """
+        Raises an AssertionError if the set of fold names in task_config doesn't
+        match the set of fold names used for splits in the metadata
+        """
+        if set(metadata["split"].unique()) != set(self.task_config["splits"]):
+            raise AssertionError(
+                "Names of splits in metadata don't match the required names for a "
+                f"kfold dataset. Expected: {self.task_config['splits']}. "
+                f"Received: {metadata['split'].unique()}."
+            )
```

Comment:
We're not using splits for kfolds, so why do we need this function?

Reply:
`Splits` correspond to the `folds` in the fold mode. So essentially there will always be a split parameter in the task config